shared/settings/advanced/index.tsx

```diff
@@ -208,6 +249,153 @@ class Developer extends React.Component<Props, State> {
   }
 }
 
+// A list so the order of the elements is fixed
+const ProxyTypeToDisplayNameList = [
+  ['noProxy', 'No Proxy'],
+  ['httpConnect', 'HTTP Connect'],
+  ['socks', 'Socks5'],
+]
+const ProxyTypeToDisplayName = ProxyTypeToDisplayNameList.reduce((obj, item) => {
+  obj[item[0]] = item[1]
+  return obj
+}, {})
+const DisplayNameToProxyType = {
+  'HTTP Connect': 'httpConnect',
+  'No Proxy': 'noProxy',
+  Socks5: 'socks',
+}
+
+type ProxyState = {
+  address: string
+  port: string
+  proxyType: string
+  certPinning: boolean
+  showCertPinningConfirmationModal: boolean
+}
+
+type ProxyProps = Props & {confirmDisableCertPinning: (disableCertPinning: () => void) => void}
+
+class ProxySettings extends React.Component<ProxyProps, ProxyState> {
+  constructor(props: ProxyProps) {
+    super(props)
+
+    this.state = {
+      address: '',
+      certPinning: true,
+      port: '',
+      proxyType: ProxyTypeToDisplayName['noProxy'],
+      showCertPinningConfirmationModal: false,
+    }
+
+    RPCTypes.configGetProxyDataRpcPromise()
```

Comment:
We don't dispatch RPCs from the display components anywhere else: the usual pattern is:

* accept an `onLoadProxyData` from the container component that calls something like `dispatch(ConfigGen.createLoadProxyData())`

* have a saga listening for that action that perform the RPC and then calls `ConfigGen.createProxyDataReceived(proxydata)`

* have a reducer that puts that into the Redux store

* have this component subscribed to the Redux store via the container's mapStateToProps

etc

Reply:
Done.